code/game/objects/effects/spawners/random/random.dm

```diff
@@ -102,10 +102,21 @@
 		for(var/turf/turf_in_view in view(radius, get_turf(src)))
 			if(isclosedturf(turf_in_view) || (isgroundlessturf(turf_in_view) && !GET_TURF_BELOW(turf_in_view)))
 				continue
+			if(!has_unblocked_line(turf_in_view))
+				continue
+
 			scatter_locations += turf_in_view
 
 	return scatter_locations
 
+/obj/effect/spawner/random/proc/has_unblocked_line(destination)
+	. = TRUE
+	var/turf/us = get_turf(src)
```

Comment:
Could use a better name like our_turf, or I think you could also just avoid the variable entirely